src/uint256.h

```diff
@@ -91,6 +93,27 @@ class base_blob
     }
 };
 
+template <unsigned int BITS>
+consteval base_blob<BITS>::base_blob(std::string_view hex_str)
+{
+    // Non-lookup table version of HexDigit().
+    auto from_hex = [](const char c) -> int8_t {
+        if (c >= '0' && c <= '9') return c - '0';
+        if (c >= 'a' && c <= 'f') return c - 'a' + 0xA;
+        if (c >= 'A' && c <= 'F') return c - 'A' + 0xA;
+
+        assert(false);
+    };
+
+    // 2 chars per byte.
```

Comment:
The purpose of code should be to express context which we couldn't convey using code, right?
Here the code already states this clearly, the comment just seems like a distraction.

Reply:
Slightly more okay if moved to the same line? https://github.com/bitcoin/bitcoin/pull/30560#discussion_r1700885770

Reply:
it's just a nit from my part, but I just see noise here, I'm not a big fan of dead comments, especially when they repeat the code

Reply:
Agree it's better to make the code self-documenting in general. My threshold for what is a useless comment seems slightly different than yours in this case.